pandas/core/missing.py

```diff
@@ -314,6 +287,73 @@ def interpolate_1d(
         return result
 
 
+def _derive_indices_of_nans_to_preserve(
+    yvalues: ArrayLike,
+    limit: Optional[int] = None,
+    limit_area: Optional[str] = None,
+    limit_direction: Optional[str] = None,
+) -> List[int]:
+    """
+    Derive the indices of NaNs that shall be preserved after interpolation
+    This function is called by `interpolate_1d` and takes the arguments with
+    the same name from there. In `interpolate_1d`, after performing the
+    interpolation, the list of indices of NaNs to preserve is used to put
+    NaNs in the desired locations.
+
+    Parameters
+    ----------
+    yvalues: ArrayLike
+        1-d array of values of the initial Series or DataFrame
+    limit: int
+    limit_area: str
+    limit_direction: str
+
+    Returns
+    -------
+    preserve_nans: list of int
+        Set of index pointers to where NaNs should be preserved in `yvalues`
+    """
+
+    invalid = isna(yvalues)
+    valid = ~invalid
+
+    # These are sets of index pointers to invalid values... i.e. {0, 1, etc...
+    all_nans = set(np.flatnonzero(invalid))
```

Comment:
better to use numpy set ops here, no?

e.g. np.unique

Reply:
not relevant anymore